src/altinn-app-frontend/src/utils/appUrlHelper.ts

```diff
@@ -186,6 +188,70 @@ export const getOptionsUrl = ({
   if (language) {
     params.language = language;
   }
+  if (formData && dataMapping) {
+    const mapped = mapFormData(formData, dataMapping);
+
+    params = {
+      ...params,
+      ...mapped,
+    };
+  }
+
+  url.search = new URLSearchParams(params).toString();
+  return url.toString();
+};
+export interface IGetDataListsUrlParams {
+  dataListId: string;
+  dataMapping?: IMapping;
+  formData?: IFormData;
+  language?: string;
+  secure?: boolean;
+  instanceId?: string;
+  pageSize?: string;
+  pageNumber?: string;
+  sortDirection?: SortDirection;
+  sortColumn?: string;
+}
+
+export const getDataListsUrl = ({
+  dataListId,
+  dataMapping,
+  formData,
+  language,
+  pageSize,
+  pageNumber,
+  sortDirection,
+  sortColumn,
+  secure,
+  instanceId,
+}: IGetDataListsUrlParams) => {
+  let url: URL;
+  if (secure) {
+    url = new URL(`${appPath}/instances/${instanceId}/datalists/${dataListId}`);
+  } else {
+    url = new URL(`${appPath}/api/datalists/${dataListId}`);
+  }
+  let params: Record<string, string> = {};
+
+  if (language) {
+    params.language = language;
+  }
+
+  if (pageSize) {
+    params.size = pageSize;
+  }
+
+  if (pageNumber) {
+    params.page = pageNumber;
+  }
```

Comment:
What if the page number is `0`? Might need to check if `pageNumber !== undefined` on this one